numba/cpython/unicode.py

```diff
@@ -2342,6 +2342,28 @@ def impl(i):
             return _PyUnicode_FromOrdinal(i)
         return impl
 
+
+@overload(str)
+def integer_str(n):
+    if isinstance(n, types.Integer):
+        def impl(n):
+            flag = False
+            if n < 0:
+                n = -n
+                flag = True
+            if n == 0:
+                return '0'
+            l = []
+            while n > 0:
+                # there is an implicit conversion of n to float here
+                c = chr(int(ord('0') + (n % 10)))
```

Comment:
Don't think the cast won't work for large values, e.g. `np.uint64(~0x0)` ?

Reply:
Think this may end up requiring something like:
```python
@overload(str)
def integer_str(n):
    ten = n(10)
    def impl(n):
        flag = False
        if n < 0:
            n = -n
            flag = True
        if n == 0:
            return '0'
        l = []
        while n > 0:
            chr_arg = ord('0') + (n % ten)
            c = chr(chr_arg)
            n = n // ten
            l.append(c)
        if flag:
            l.append('-')
        return ''.join(l[::-1])
    return impl
```

Reply:
Thanks, I think this last commit solves this problem